### PR TITLE
Added Metric and ValidationDataset classes

### PR DESCRIPTION
**Pull request by i-khokhlov**
# Usage

## Calc metrics
```
from DDT.utils import Metric
from DDT.utils import ValidationDataset

metrics_all = []
m = Metric()

for val_dict in tqdm(val):
    model.eval()
    device = "cuda:0"
    with torch.no_grad():
        pred_depth = model.depth(val_dict["tensor"].to(device, dtype=torch.float))
    depth_image = pred_depth[0].detach().cpu().permute(1, 2, 0).numpy()[:, :, 0]
    metrics_all.append(m.calc_metrics(depth_image, cv2.resize(val_dict["groundtruth_depth"], (384, 128))))
    
for el1, el2 in zip(m.get_header(), np.array(metrics_all).mean(axis=0)):
    print(el1, el2) 
```

## View results on one image
```
import matplotlib.pyplot as plt
i = 20
val_dict = val[i]
model.eval()
device = "cuda:0"
with torch.no_grad():
    pred_depth = model.depth(val_dict["tensor"].to(device, dtype=torch.float))


depth_image = pred_depth[0].detach().cpu().permute(1, 2, 0).numpy()[:, :, 0]

m = Metric()
metrics_calc = m.calc_metrics(depth_image, cv2.resize(val_dict["groundtruth_depth"], (384, 128)))
for el1, el2 in zip(m.get_header(), metrics_calc):
    print(el1, el2)


figure = plt.figure(dpi=200)

plt.subplot(2, 2, 2)
plt.imshow(val_dict["tensor"].permute(2, 3, 1, 0).detach().cpu().numpy()[:,:,:,0])
plt.title("Image transformed")
plt.xticks([])
plt.yticks([])

plt.subplot(2, 2, 1)
plt.imshow(val_dict["image"])
plt.title("Image")
plt.xticks([])
plt.yticks([])

plt.subplot(2, 2, 3)
plt.imshow(np.clip(1.0 / depth_image, 0, 100) / 100, cmap="inferno")
plt.title("Predicted depth")
plt.xticks([])
plt.yticks([])

plt.subplot(2, 2, 4)
plt.imshow(val_dict["groundtruth_depth"], cmap="inferno")
plt.title("Groundtruth depth")
plt.xticks([])
plt.yticks([])

m = Metric()
metrics_calc = m.calc_metrics(depth_image, cv2.resize(val_dict["groundtruth_depth"], (384, 128)))
for el1, el2 in zip(m.get_header(), metrics_calc):
    print(el1, el2)

```

### CODE REVIEW

File: DDT/utils/mono_dataset.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            transforms.Resize((128, 384)),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=(0.485, 0.456, 0.406), std=(0.229, 0.224, 0.225)), # took form albumentations norm default values
+            ]) 
+    
+    def __getitem__(self, id):
+        path4depth = os.path.join(os.curdir, self.main_folder, self.depth[id])
+        path4rgb = os.path.join(os.curdir, self.main_folder, self.rgb[id])
+        img = cv2.imread(path4rgb)
+        groundtruth_dict = {'image' : img, 'tensor': self.transform(img)[None], 'groundtruth_depth': 255 - cv2.imread(path4depth, 0)}
```
Comment (kshpv) [inline DDT/utils/mono_dataset.py:33]: 'groundtruth_depth': 255 - cv2.imread(path4depth, 0)} - is range from 0 to 255?

Comment (i-khokhlov) [reply DDT/utils/mono_dataset.py:33]: Yep. For example, in TUM-RGBD sequence 3 range is from 63 to 255 for all ground truth depth images.
